src/runtime_src/hip/api/hip_memory.cpp

```diff
@@ -215,6 +205,27 @@ namespace xrt::core::hip
     free(host_src);
```

Comment:
This one caught my attention.  What happens if `write` throws?   Never use new/delete, alloc/free.
Consider using xrt_core::aligned_alloc (core/common/memalign.h) which manages the memory (RAII).

Reply:
fixed.